src/core/mac/mac.cpp

```diff
@@ -446,17 +446,25 @@ void Mac::SetSupportedChannelMask(const ChannelMask &aMask)
     return;
 }
 
-otError Mac::SetNetworkName(const char *aBuffer, uint8_t aLength)
+otError Mac::SetNetworkName(const char *aNameString)
 {
-    otError error  = OT_ERROR_NONE;
-    uint8_t newLen = static_cast<uint8_t>(strnlen(aBuffer, aLength));
+    NetworkName::Data data(aNameString, NetworkName::kMaxSize + 1);
```

Comment:
Does `data.mLength` include the NULL character? Should we have the `+ 1` here?

Reply:
Yes.  `data.mLength` does not include/count the null char.
But the `+1` here is intentional to ensure we reject a `aNameString` which has a length longer than `kMaxSize`.

- Basically `NetworkName::Data` has no restriction on what its `mLength` can be set to. 
- When setting from `aNameString` we only care about `kMaxSize + 1` chars in the string.
- When we use `SetNetworkName(data)` (which in turn calls `NetworkName::Set(data)` method), the `Set()` method will check for new length to be fit in `kMaxSize` (otherwise retursn `OT_ERROR_INVALID_ARGS`)
   -  `NetworkName::Set()` will check for any NULL chars within `data` and calculates the `newLen` for the name.

Reply:
Makes sense. Do you mind adding a short comment on why the `+ 1`?